### PR TITLE
Basic deployment docs

### LINKED ISSUE #1810: Add some deployment docs
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates
- [x] This is a request for a change to documentation
- [x] This isn't a feature request or bug report
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :ledger: Where is the problem?

Following the merge of #1779 we lost a lot of documentation. We should start with some new deployment documentation using `data_safe_haven/README.md` as a basis.

## :memo: Required documentation changes

Write a deployment guide

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [x] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).

### :arrow_heading_up: Summary

Add some basic deployment and management docs.

### :closed_umbrella: Related issues

Closes #1810

### :microscope: Tests

Tested on a local build.

PR build is [here](https://data-safe-haven--1867.org.readthedocs.build/en/1867/)

### CODE REVIEW

File: docs/source/deployment/deploy_sre.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Make sure that your config file contains one or more SRE sections.
+
+:::{tip}
+You can check this by running: `dsh config show` and looking at the `sres` section.
+:::
+
+## Requirements
+
+:::{important}
+As private endpoints for flexible PostgreSQL are still in preview, the following command is currently needed:
```
Comment (craddm) [inline docs/source/deployment/deploy_sre.md:18]:
```suggestion
As private endpoints for flexible PostgreSQL are still in preview, the following command is currently needed if you are intending to deploy a PostgreSQL database:
```

File: docs/source/management/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+```{code} shell
+$ dsh admin register-users -s <SRE name> <username1> <username2>
+```
+
+where you must specify the usernames for each user you want to add to this SRE
+
+## Listing available users
+
+1. You can do this in your Entra tenant by browing to `Identity > Users > All users.`.
```
Comment (craddm) [inline docs/source/management/index.md]:
```suggestion
1. You can do this in your Entra tenant by browsing to `Identity > Users > All users.`.
```

File: docs/source/deployment/deploy_sre.md (hunk 1 above)
Comment (jemrobinson) [reply docs/source/deployment/deploy_sre.md:18]: As we also use flexible PostgreSQL to back e.g. Gitea and HedgeDoc I don't think there's any way around this.

**Review by craddm — APPROVED** [review]
LGTM

**Pull request merged**